plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -214,43 +212,56 @@ func initQueries(s *SQLServer) error {
 
 // Gather collect data from SQL Server
 func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
-	if !s.isInitialized {
-		if err := initQueries(s); err != nil {
-			acc.AddError(err)
-			return err
+	var wg sync.WaitGroup
+
+	for _, pool := range s.pools {
+		for _, query := range s.queries {
+			wg.Add(1)
+			go func(pool *sql.DB, query Query) {
+				defer wg.Done()
+				acc.AddError(s.gatherServer(pool, query, acc))
+			}(pool, query)
 		}
 	}
 
+	wg.Wait()
+	return nil
+}
+
+// Start initialize a list of connection pools
+func (s *SQLServer) Start(acc telegraf.Accumulator) error {
```

Comment:
I see this func only used by unit test and not called from plugin. Is that intenional?